packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -0,0 +1,128 @@
+import retrying
+
+
+LATENCY = 60
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+def test_metrics_ping(cluster):
+    """ Test that the metrics service is up.
+    """
+    for agent in cluster.slaves:
+        response = cluster.metrics.get('ping', node=agent)
+        assert response.status_code == 200, 'Got non-200 status code: {}'.format(response.content)
+        'from private agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on private, got:\n{}'.format(response.content)
+        'agent.'
+
+    for agent in cluster.public_slaves:
+        response = cluster.metrics.get('ping', node=agent)
+        assert response.status_code == 200, 'Got non-200 status code: {}'.format(response.content)
+        'from public agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on public agent, got:\n{}'.format(response.content)
+
+    for master in cluster.masters:
+        response = cluster.metrics.get('ping', node=master)
+        assert response.status_code == 200, 'Got non-200 status code {}'.format(response.content)
+        'from public agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on master, got:\n{}'.format(response.content)
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+def test_metrics_node(cluster):
+    """Test /system/metrics/api/v0/node endpoint returns
+    correct metrics.
+    """
+    expected_node_metric_datapoints = [
+        'uptime',
+        'processes',
+        'cpu.cores',
+        'load.1min',
+        'load.5min',
+        'load.15min',
+        'cpu.total',
+        'cpu.user',
+        'cpu.system',
+        'cpu.idle',
+        'cpu.wait',
+        'memory.total',
+        'memory.free',
+        'memory.buffers',
+        'memory.cached',
+        'swap.total',
+        'swam.free',
+        'swap.used']
+
+    def expected_datapoint_response(response):
+        for datapoint in response["datapoints"]:
+            if datapoint["name"] not in expected_node_metric_datapoints:
+                return False
+            return True
+
+    for agent in cluster.slaves:
+        response = cluster.metrics.get('node', node=agent)
+
+        assert response.status_code == 200, 'Got non-200 status code '
```

Comment:
assert r.status_code == 200 ends up producing really hard to debug / track down errors when it fails (you don't get the status code or response body printed which are both usually critical in solving). Please use r.raise_for_status() or do a `assert r.ok, "status_code: {}, content: {}".format(r.status_code, r.content))`. This goes for this instance and all the other ones. The "Got non-200 status code" is also not very informative when this fails.